lib/fs/basicfs_lstat_windows.go

```diff
@@ -77,10 +77,13 @@ func (fi *dirJunctFileInfo) IsDir() bool {
 var junctionPointModeMask os.FileMode
 
 func init() {
+	// Per https://tip.golang.org/doc/go1.23#minor_library_changes
+	// In go1.22 (and go1.23 when GODEBUG=winsymlink=0) directory junctions
+	// (aka "mount points") always have ModeSymlink set.
 	junctionPointModeMask = os.ModeSymlink
```

Comment:
Why are we keeping mode symlink in the mask for go 1.23? The linked changelog says "Mount points no longer have [ModeSymlink](https://tip.golang.org/pkg/os#ModeSymlink) set,", so I'd expect the mask be set to only os.ModeIrregular for >= 1.23 and to only os.ModeSymlink on < 1.23.

Reply:
@imsodin Agreed, corrected in https://github.com/syncthing/syncthing/pull/9786/commits/69f9c95028828fd0c11df73048aaf5230de431d2

Reply:
@imsodin Because users can set `GODEBUG=winsymlink=0` in their environment, which will cause Syncthing to operate in go1.22 mode (where `ModeSymlink` is set). See https://go.dev/doc/godebug#history